Update the letter validation messages.

servingUpAces opened the pull request:
# What
This PR updates the way the validation errors are displayed for precompiled letters sent in via the API. They now use "partials/check/letter-validation-failed-banner.html", which is the same banner that is displayed when an uploaded letter failed validation.
Also added a new error message for missing address blocks.
There is an outstanding question about what the page title should be, the standard is to match the H1 on the page with the page title, this PR has changed the H1 from `Letter` to the first line of the error message. We'd like to take some time to answer this question, therefore, leaving the page titles as is for now. 

## Before
![Screenshot 2020-01-09 at 10 27 39](https://user-images.githubusercontent.com/4282990/72145199-ed232780-3391-11ea-8f10-13b59418d3a7.png)

# After
### Letter outside content area
![Screenshot 2020-01-10 at 10 10 02](https://user-images.githubusercontent.com/4282990/72145102-a9c8b900-3391-11ea-8530-dbf0b9705153.png)

### Letter that is missing EOF tag
![Screenshot 2020-01-10 at 10 05 22](https://user-images.githubusercontent.com/4282990/72145107-acc3a980-3391-11ea-996b-19a7c3b77622.png)

### Cancelled letters now show message using "govuk-warning-text"
![Screenshot 2020-01-10 at 10 04 13](https://user-images.githubusercontent.com/4282990/72145171-daa8ee00-3391-11ea-8da6-73bb01e4a3c2.png)
 
### Added a new error message for a letter that is missing an address block.
![Screenshot 2020-01-10 at 10 14 56](https://user-images.githubusercontent.com/4282990/72145274-12b03100-3392-11ea-81d0-d3f0bb017f61.png)

Pair: becca&karl

Commit:
Update the letter validation messages.

Added a new erorr message for a letter that is missing an address block.

Pair: becca&karl

klssmith commented on app/utils.py:
```diff
@@ -566,14 +566,14 @@ def get_letter_printing_statement(status, created_at):
-        'title': 'We cannot print your letter',
-        'detail': 'The content appears outside the printable area on {invalid_pages}.<br>'
+        'title': 'Your content is outside the printable area',
+        'detail': 'You need to edit {invalid_pages}.<br>'
```
We don't show the title for letters sent through the api (only for uploaded letters), so for letters sent through the api the error message has changed from `Validation failed. The content appears outside the printable area on page 1.` to `Validation failed. You need to edit page 1.` which doesn't explain what the issue is any more.

Not sure if that's an issue or not since we still show the overlay, but it might be less clear what's wrong with the letter.

servingUpAces replied:
It's worth taking sometime to review the overlay page for api uploads.

Commit:
Update the way a validation error displays for a letter.

There is an outstanding question about what the page title should be. For the time being we left it as Letter. But we will take a proper look at this in another story.
Use govuk-warning-text for cancelled letters.
Removed the test for permanent-failures since that's not a valid status type for letters.

servingUpAces commented:
I've created a new PR to add the new content changes, https://github.com/alphagov/notifications-admin/pull/new/new-no-address-message without changing the existing format. It seems a longer conversation is required before implementing a change to the validation errors on precompiled letters uploaded by the API.

quis commented:
Closing in favour of https://github.com/alphagov/notifications-admin/pull/3254
